### PR TITLE
add set_location method for setting geolocation

### PR DESCRIPTION
**Pull request by mclark**
This just hooks into web driver's location context module and then calls the existing methods on the appium server to set the device's gps coordinates.

Not sure the best way to write a test for this; I'm open to suggestion.

### CODE REVIEW

**Comment by bootstraponline** [comment]
I think we need get_location and then at that point testing this is relatively easy?

**Comment by mclark** [comment]
agreed, but it's not implemented on the appium server, so that would be a separate pull request unfortunately.  I'm also inexperienced with node.js.

**Comment by bootstraponline** [comment]
In that case I think simply calling set_location in the existing android tests is sufficient. At least we'll know if it raises an exception.

**Comment by bootstraponline** [comment]
I tried a simple test in ARC.

```ruby
d = driver.driver
d.extend Selenium::WebDriver::DriverExtensions::HasLocation
d.set_location(1,2,3)
Selenium::WebDriver::Error::UnknownError: ERROR running Appium command: port should be a number or string: false
```

```
info: [debug] Sending telnet command to device: geo fix 2 1
info: [debug] Getting running emulator port
info: <-- POST /wd/hub/session/47d3f15b-e2d5-419b-b7a3-13986b293f98/location 500 12.303 ms - 94 
TypeError: port should be a number or string: false
```

Is this only intended to work on emulators and not real devices? If so this needs to be documented in the [yard doc](yardoc.org).

**Comment by mclark** [comment]
Ah yes of course, good catch!  Thanks!  I will add this and the noted test.

**Comment by bootstraponline** [comment]
In the appium server there's a special test app that's able to verify the location was [set correctly](https://github.com/appium/appium/blob/d922710a39a591e9cdddf078c8b1f27651ab0490/test/functional/ios/testapp/location-specs.js#L19) so I don't think we need to duplicate that effort. A simple test that invokes the method will detect Ruby specific issues such as if they method is deleted or renamed. If you have questions about this, let me know. 

Also I think this method would be better off using a hash.

```ruby
def set_location(opts = {})
  latitude = opts.fetch(:latitude)
  longitude = opts.fetch(:longitude)
  altitude = opts.fetch(:altitude, 75)
  @driver.set_location(latitude, longitude, altitude)
end
```

then the test would be:

```ruby
t 'set_location' do # note this should be added in both Android/iOS tests 
  set_location latitude: 1, longitude: 2, altitude: 3
end
```

**Comment by bootstraponline** [comment]
also since the ruby_lib tests run on both emulators/physical devices, the android version of the test will have to rescue that exception.

**Comment by mclark** [comment]
ok I'll make the changes you've asked for, but running the ios tests as instructed in the readme I get a hundred undefined method errors like the following:
`rake install`
`rake ios`

```
89) Error:
driver::methods#test_0001_status:
NameError: undefined local variable or method `appium_server_version' for #<#<Class:0x007fa37ba37e08>:0x007fa37c390f40>
    /Users/matt/Projects/ruby_lib/android_tests/lib/android/specs/driver.rb:118:in `block (3 levels) in <top (required)>'
```

is there something I am missing?

**Comment by bootstraponline** [comment]
pull from master, that issue should be fixed with #336

**Comment by mclark** [comment]
changes made and squashed, back to you @bootstraponline

File: android_tests/lib/android/specs/driver.rb [hunk 1]
```diff
@@ -175,6 +175,17 @@ def validate_path(path)
...
     end
 
+    # simple integration sanity test to check for unexpected exceptions
+    t 'set_location' do
+      begin
+        set_location latitude: 55, longitude: -72, altitude: 33
+      rescue Selenium::WebDriver::Error::UnknownError => e
+        # on android this method is expected to fail with this message when running
+        # on a regular device, or on genymotion.
+        raise unless e.message.start_with?('ERROR running Appium command: port should be > 0 and < 65536')
```
Comment (bootstraponline) [inline android_tests/lib/android/specs/driver.rb]: I think this should be `raise e`?

**Comment by bootstraponline** [comment]
The code looks great. One more comment, we should raise the original exception instead of just a generic `raise`.

**Comment by bootstraponline** [comment]
I checked in pry and it looks like by default raise will do the right thing. Thanks for the pull request!

**Comment by mclark** [comment]
Happy to help!  Thanks for the review, feedback, and quick merge on the PR!

**Pull request merged**